variables.tf

```
@@ -1,103 +1,76 @@
-variable "ipam" {
-  description = "Object of inputs for IPAM resources"
+variable "vpc_ipam" {
+  description = "Object of inputs for all IPAM configuration resources"
   type = object({
-    region_name = optional(string)
-    tags        = optional(map(string))
-    cascade     = optional(bool)
+    ipam = optional(object({
+      operating_regions = list(object({
+        region_name = optional(string) #required
```

if it's required, remove the optional marker...

```suggestion
        region_name = string #required
```